src/components/Timeline/context/Context.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { createContext, useContext } from 'react';
+import {
+    CollapseProviderType,
+    defaultTimelineContextValue,
+    TimelineContextState,
+} from './types';
+
+export const TimelineProvider: CollapseProviderType = ({
+    children,
+    isOppositeContent,
+}) => {
+    return (
+        <TimelineContext.Provider
+            value={{
+                isOppositeContent,
```

Comment:
Nová reference při každém přerenderování Providera přerenderuje všechny elementy, které na to naslouchají, `value` bych pro jistotu memoizoval, pokud to má být objekt. Jestli jde jenom o ten `isOppositeContent`, value může zůstat přímo jeho hodnota, potom se nemusí memoizovat nic